### ISSUE TITLE
[QUESTION] Playwright Docker container creates root owned files on the host machine

### ISSUE THREAD

**Issue by cgmonte** [microsoft/playwright#28427]
Hi Playwright people, thanks for the amazing tool!

I am just starting to use it and I am trying to run the tests on a local docker container. 

Tests run perfectly, but I am having a minor issue. Because tests are running as root inside the container, report, results etc. are created as owned by the root user. Some users may not have sudo permissions and, when running the tests locally (without docker), the test session will fail because those files cannot be replaced/updated.

Question: how can I solve this? I tried to make Playwright create its files inside the container by playing with the `outputDir` property, but it always creates on the host. Any idea is welcome.

### System info
- Playwright Version: [v1.39]
- Operating System: [Fedora 37]
- Browser: [Chromium]
- Other info:

### Source code

**Docker run command**:

`docker run -v /localpath:/mnt/host --network="host" --rm -it playwright:localbuild-focal /bin/bash -c 'cd /mnt/host && npx playwright test'`

- [x] I provided exact source code that allows reproducing the issue locally.

**Config file**

```js
module.exports = defineConfig({
	forbidOnly: !!process.env.CI,
	projects: [
		{
			name: 'chromium',
			use: {...devices['Desktop Chrome']},
		},
	],
	reporter: [
		[
			'html',
			{
				open: 'never',
			},
		],
		[
			'junit',
			{
				outputFile: 'test-results/TEST-playwright.xml',
			},
		],
	],
	retries: process.env.CI ? 2 : 0,
	testDir: './tests',
	timeout: 60000,
	use: {
		baseURL: process.env.PORTAL_URL
			? process.env.PORTAL_URL
			: 'http://localhost:8080',
		trace: 'on-first-retry',
	},
	workers: process.env.CI ? 3 : undefined,
});
```

**Comment by pavelfeldman** [comment]
Additional info:

*User Namespaces*: Docker supports user namespace remapping, which can map the root user inside a container to a non-root user on the host. This helps mitigate the risk of a container process gaining root-level access to the host system.

*Run Containers as Non-Root*: You can configure Docker containers to run as a non-root user. By specifying a user in the Dockerfile or using the --user flag when starting a container, you can limit the privileges of processes inside the container.

**Comment by cgmonte** [comment]
Yeah, adding `--user pwuser` to `docker run` works just fine! Thank you.
